slobrok/src/vespa/slobrok/server/map_view.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "map_view.h"
+#include <vespa/log/log.h>
+LOG_SETUP(".slobrok.map_view");
+
+namespace slobrok {
+
+MapView::MapView()
+  : _map(),
+    _currGen(0),
+    _lock()
+{}
+
+MapView::~MapView() {
+    clear();
```

Comment:
This implies that we are ok with adding listeners that are never removed. Should we be ok with that?

Reply:
> This implies that we are ok with adding listeners that are never removed. Should we be ok with that?

it's an option to assert that listener list is empty instead, will revisit when we get it wired in.